### PR TITLE
Add compression benchmark and refactor serialization

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/602

### CODE REVIEW

**Comment by zsybupt** [comment]
I think we can use the same implementation for compress message

**Comment by JamesNK** [comment]
Sure. There are going to be some changes to serialization in the next day or two, I'll look at it at the same time.

File: src/Grpc.AspNetCore.Server/Internal/ReadOnlySequenceStream.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    // Potentially remove in the future when https://github.com/dotnet/corefx/issues/31804 is implemented
+    internal class ReadOnlySequenceStream : Stream
+    {
+        private static readonly Task<int> TaskOfZero = Task.FromResult(0);
```
Comment (JunTaoLuo) [inline src/Grpc.AspNetCore.Server/Internal/ReadOnlySequenceStream.cs]: const?

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/Internal/ReadOnlySequenceStream.cs]: `Task` not a primitive type so const is not allowed.

**Pull request merged**